cmake/SpectreTargetSources.cmake

```diff
@@ -26,37 +26,11 @@
 # - EXTRA_ARGS: the sources, PUBLIC/PRIVATE/INTERFACE keywords, and generator
 #               expressions to set the sources to.
 function(spectre_target_sources TARGET)
-  if(POLICY CMP0076)
-    # New behavior is available, so just forward to it by ensuring
-    # that we have the policy set to request the new behavior, but
-    # don't change the policy setting for the calling scope
-    cmake_policy(PUSH)
-    cmake_policy(SET CMP0076 NEW)
-    target_sources(${TARGET} ${ARGN})
-    cmake_policy(POP)
-    return()
-  endif()
-
-  # Must be using CMake 3.12 or earlier, so simulate the new behavior
-  unset(_SOURCE_FILES)
-  get_target_property(_TARGET_SOURCE_DIR ${TARGET} SOURCE_DIR)
-
-  foreach(SOURCE_FILE ${ARGN})
-    string(FIND "${SOURCE_FILE}" "$<" GENERATOR_EXPR_START_INDEX)
-    set(_IS_GENERATOR_EXPR OFF)
-    if(${GENERATOR_EXPR_START_INDEX} EQUAL 0)
-      set(_IS_GENERATOR_EXPR ON)
-    endif(${GENERATOR_EXPR_START_INDEX} EQUAL 0)
-
-    if(NOT SOURCE_FILE STREQUAL "PRIVATE" AND
-        NOT SOURCE_FILE STREQUAL "PUBLIC" AND
-        NOT SOURCE_FILE STREQUAL "INTERFACE" AND
-        NOT IS_ABSOLUTE "${SOURCE_FILE}" AND
-        NOT _IS_GENERATOR_EXPR)
-      # Absolute path to source
-      set(SOURCE_FILE "${CMAKE_CURRENT_LIST_DIR}/${SOURCE_FILE}")
-    endif()
-    list(APPEND _SOURCE_FILES ${SOURCE_FILE})
-  endforeach()
-  target_sources(${TARGET} ${_SOURCE_FILES})
+  # New behavior is available, so just forward to it by ensuring
+  # that we have the policy set to request the new behavior, but
+  # don't change the policy setting for the calling scope
+  cmake_policy(PUSH)
+  cmake_policy(SET CMP0076 NEW)
+  target_sources(${TARGET} ${ARGN})
+  cmake_policy(POP)
```

Comment:
we don't need this anymore I think. The "new" policy (now always available and set to "new") is to convert relative to absolute paths. So just call `target_sources` and nothing else.

Please update the function docs.

Reply:
When did the behavior change?  I don't see anything referencing versions between 3.13 and 3.18 in the CMake docs.

Reply:
Changed in 3.13: https://cmake.org/cmake/help/latest/policy/CMP0076.html

Reply:
This means the policy is set to "new" automatically for projects with a min cmake version >= 3.13

Reply:
Ah, I see.  It's a nonlocal thing.  I'll update it.